contracts/src/v0.8/functions/dev/v1_0_0/FunctionsBilling.sol

```diff
@@ -218,26 +211,26 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
       callbackGasLimit: request.callbackGasLimit,
       estimatedTotalCostJuels: estimatedTotalCostJuels,
       timeoutTimestamp: uint32(block.timestamp + config.requestTimeoutSeconds),
-      requestId: requestId,
+      requestId: bytes32(0), // requestId is computed after hashing the commitment data
       donFee: donFee,
       gasOverheadBeforeCallback: config.gasOverheadBeforeCallback,
       gasOverheadAfterCallback: config.gasOverheadAfterCallback
     });
 
-    s_requestCommitments[requestId] = keccak256(abi.encode(commitment));
+    commitment.requestId = _computeRequestId(commitment, request.initiatedRequests + 1);
+
+    s_requestCommitments[commitment.requestId] = keccak256(abi.encode(commitment));
```

Comment:
Why do you need a different value here than the request ID? You now do a very expensive abi encode twice.

Reply:
in CCIP we use very similar logic, but we don't abi.encode and hash again. We use the request ID as identifier throughout the code. Hashing it again with the commitment doesn't give any additional security either

Reply:
Thanks!  We wanted to hash the whole commitment to ensure if any commitment data changed that the requestId would be unique, but we discussed an alternate approach which should provide the same guarantees without wasting gas.